### ISSUE TITLE
Use QuickCheck-like approach to Flammable?

### ISSUE THREAD

**Issue by paf31** [sharkdp/purescript-flarecheck#1]
Perhaps one approach to removing the `uncurry` calls is to use a `Testable`-like class:

``` purescript
class Flammable a

-- Base types
instance Interactive String
instance Interactive Boolean
instance Interactive ...

instance (Flammable a, Interactive b) => Interactive (a -> b)
```

The trade-off is that you need an `Interactive` instance for every base type, unless you want to add something like `Show a => Interactive a`, which might lead to unpleasant overlapping instance problems.

**Comment by sharkdp** [comment]
That's what I tried initially, but I did not manage to write this instance:

``` purs
instance (Flammable a, Interactive b) => Interactive (a -> b)
```

Assuming you want to keep the definition of `Interactive`:

``` purs
class Interactive t where
  createUI :: forall e. t -> UI e String
```

I don't know how to write

``` purs
createUI :: forall e. (a -> b) -> UI e String
createUI f = ???
```

I can write `f <$> spark`, which gives me a `UI e b`, but I don't know how to transform it to `UI e String` because `UI` is not `Monad` (otherwise I don't know how to use the `createUI` for `b`).

**Comment by sharkdp** [comment]
Oh, I just saw your pull request...

**Comment by sharkdp** [comment]
Closed via #2
